fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -884,6 +890,42 @@ private void checkInteractionModel(final UpdateRequest request) {
         }
     }
 
+    /**
+     * Check sparql statements for disallowed modifications
+     * @param request
+     */
+    private void checkForDisallowedTriples(final UpdateRequest request) {
+        final List<Quad> quads = new ArrayList<>();
+
+        for (final Update operation : request.getOperations()) {
+            if (operation instanceof UpdateModify) {
+                final UpdateModify op = (UpdateModify) operation;
+                quads.addAll(op.getDeleteQuads());
+                quads.addAll(op.getInsertQuads());
+            } else if (operation instanceof UpdateData) {
+                final UpdateData op = (UpdateData) operation;
+                quads.addAll(op.getQuads());
+            } else if (operation instanceof UpdateDeleteWhere) {
+                final UpdateDeleteWhere op = (UpdateDeleteWhere) operation;
+                quads.addAll(op.getQuads());
+            }
+        }
+
+        quads.stream().forEach(e -> {
+           ensureNoMementNamespacedObjects(e.asTriple());
```

Comment:
This is probably fine for now, but should likely be refactored to not be re-iterating through the request for every triple that we want to block. 
For example, we are doing something quite similar [directly above this method](https://github.com/fcrepo4/fcrepo4/pull/1437/files#diff-7aee50d5030524028026aa94cd75f350R855).